src/main/java/com/coveros/selenified/element/check/Excludes.java

```diff
@@ -104,18 +103,26 @@ default String checkClazz(String unexpectedClass, double waitFor, double timeToo
      * @param timeTook  - the amount of time it took for wait for something (assuming we had to wait)
      * @return String[]: all of the attributes of the element. null will be returned if the element isn't present
      */
-    default String[] checkAttribute(String attribute, double waitFor, double timeTook) {
+    default Set<String> checkAttribute(String attribute, double waitFor, double timeTook) {
         // record the action and get the attributes
-        String[] allAttributes = getAttributes(attribute, "without", waitFor);
+        Map<String, String> atts = getElement().get().allAttributes();
+        Set<String> allAttributes = new HashSet<>();
+        if (atts != null) {
+            allAttributes = atts.keySet();
+        }
         // record the result
-        if (allAttributes == null || Arrays.asList(allAttributes).contains(attribute)) {
-            getOutputFile().recordActual(getElement().prettyOutputStart() + " contains the attribute of <b>" + attribute + "</b>",
-                    timeTook, Success.FAIL);
+        if (atts == null || allAttributes.contains(attribute)) {
+            getReporter().fail(getElement().prettyOutput() + " without attribute <b>" + attribute + "</b>", waitFor, getElement().prettyOutputStart() + " contains the attribute of <b>" + attribute + "</b>",
+                    timeTook);
+        } else {
+            getReporter().pass(getElement().prettyOutput() + " without attribute <b>" + attribute + "</b>", waitFor, getElement().prettyOutputStart() + " does not contain the attribute of <b>" + attribute + "</b>" +
+                    ONLY_VALUE + String.join(", ", allAttributes) + "</b>", timeTook);
+        }
+        if (atts == null) {
+            return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1168)